smdebug/core/reader.py

```diff
@@ -54,3 +57,23 @@ def read_tensors(self, check="minimal"):
         if check is True:
             check = "minimal"
         return self._reader.read_tensors(check=check)
+
+    # Read the events that match the given regex list. If Regex list is not specified all events are read.
+    def read_events(self, check="minimal", regex_list=None):
+        if check is True:
+            check = "minimal"
```

Comment:
is it okay to have such weakly typed semantics, where we are changing the type of the variable from boolean to string?

Reply:
Agree with @anirudhacharya We should avoid this kind of pattern.

Reply:
@anirudhacharya @larroy 
Thanks for catching it!
Will fix it.

Reply:
Addressed the review comments.